src/nvidia_resiliency_ext/fault_tolerance/_ft_rendezvous.py

```diff
@@ -874,16 +892,61 @@ def _remove_from_redundancy_list(self) -> None:
 
     @staticmethod
     def _assign_ranks(
-        participants: Dict[_NodeDesc, int], prev: Dict[_NodeDesc, int]
+        participants: Dict[_NodeDesc, int],
+        prev: Dict[_NodeDesc, int],
+        use_infra_group_rank: bool = False,
     ) -> Dict[_NodeDesc, int]:
-        # Assign ranks. Re-use assigment from the previous round as much as possible
+        """
+        Assign ranks to participants in the rendezvous.
+
+        Behavior depends on use_infra_group_rank and previous assignments:
+
+        1. If use_infra_group_rank=True AND prev is empty (first rendezvous):
+           - Use infrastructure ranks directly from SLURM_PROCID or GROUP_RANK
+           - Validates that all ranks are in range [0, world_size) and unique
+
+        2. If prev is not empty (subsequent rendezvous, including after failures):
```

Comment:
This needs to be done at the node level and not at individual rank level. So "rank blocks" will have to be removed and filled.